_drafts/PR-blogs/planning-docs/example-code/animated-dots/original/script.js

```diff
@@ -0,0 +1,170 @@
+    dotsHolder.push(new dots());
+  }
+
+  /*REQUEST ANIMATION FRAME*/
+  (function () {
+    var lastTime = 0;
+    var vendors = ["ms", "moz", "webkit", "o"];
+    for (var x = 0; x < vendors.length && !window.requestAnimationFrame; ++x) {
+      window.requestAnimationFrame =
+        window[vendors[x] + "RequestAnimationFrame"];
+      window.cancelAnimationFrame =
+        window[vendors[x] + "CancelAnimationFrame"] ||
+        window[vendors[x] + "CancelRequestAnimationFrame"];
+    }
+
+    if (!window.requestAnimationFrame)
+      window.requestAnimationFrame = function (callback, element) {
+        var currTime = new Date().getTime();
+        var timeToCall = Math.max(0, 16 - (currTime - lastTime));
+        var id = window.setTimeout(function () {
+          callback(currTime + timeToCall);
+        }, timeToCall);
+        lastTime = currTime + timeToCall;
+        return id;
+      };
+
+    if (!window.cancelAnimationFrame)
+      window.cancelAnimationFrame = function (id) {
+        clearTimeout(id);
+      };
+  })();
+}
+function canvasResize(event) {
+  canvas.height = window.innerHeight;
+  canvas.width = window.innerWidth;
+  cancelAnimationFrame(animate);
+}
+function MouseUp(event) {
+  if (mouseMove) {
+    mouseMove = false;
+  }
+  if (mouseDown) {
+    mouseDown = false;
+  }
+}
+function MouseDown(event) {
+  mouseDown = true;
+}
+function MouseMove(event) {
+  mouse.x = event.pageX - canvas.offsetLeft;
+  mouse.y = event.pageY - canvas.offsetTop;
+  if (mouseMove) {
+    context.lineTo(mouseX, mouseY);
+    context.stroke();
+  }
+}
+function dots() {
+  this.xPos = Math.random() * canvas.width;
+  this.yPos = Math.random() * canvas.height;
+  this.color = colors[Math.floor(Math.random() * colors.length)];
+  this.radius = Math.random() * 10;
+  this.vx = Math.cos(this.radius);
+  this.vy = Math.sin(this.radius);
+  this.stepSize = Math.random() / 10;
+  this.step = 0;
+  this.friction = 7;
+  this.speedX = this.vx;
+  this.speedY = this.vy;
+}
+
+dots.draw = function () {
+  context.clearRect(0, 0, canvas.width, canvas.height);
+  for (var c = 0; c < dotsHolder.length; c++) {
+    dot = dotsHolder[c];
+    context.beginPath();
+    distanceX = dot.xPos - mouse.x;
+    distanceY = dot.yPos - mouse.y;
+    var particleDistance = Math.sqrt(
+      distanceX * distanceX + distanceY * distanceY
+    );
+    var particleMouse = Math.max(
+      Math.min(75 / (particleDistance / dot.radius), 7),
+      1
+    );
+    context.fillStyle = dot.color;
+    dot.xPos += dot.vx;
+    dot.yPos += dot.vy;
+    if (dot.xPos < -50) {
+      dot.xPos = canvas.width + 50;
+    }
+    if (dot.yPos < -50) {
+      dot.yPos = canvas.height + 50;
+    }
+    if (dot.xPos > canvas.width + 50) {
+      dot.xPos = -50;
+    }
+    if (dot.yPos > canvas.height + 50) {
+      dot.yPos = -50;
+    }
+    context.arc(
+      dot.xPos,
+      dot.yPos,
+      (dot.radius / 2.5) * particleMouse,
+      0,
+      2 * Math.PI,
+      false
+    );
```

Comment:
`context.arc` is a function with a fairly lengthy list of positional arguments. If you work with the `context` API a lot then you might be able to remember them off the top of your head, but otherwise you have to lean on the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/arc) to figure out what they are. Or maybe if you're a diligent dev with the right IDE tools installed, you might have the API reference handy right in your code editor.

To make it easier for the rest of us, though, we can leverage some nice ES6 features to make this easier to understand at a glance: 
- The [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax), combined with
- [Object.values](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values)

This allows us to define these positional arguments as named key/value pairs in an object, but still pass the values of that object to the `context.arc` function as positional arguments. That way, we can get an at-a-glance understanding of what the values actually configure for this function without having to reference any docs.

```js
const arcConfig = {
  x: dot.xPos,
  y: dot.yPos,
  radius: (dot.radius / 2.5) * particleMouse,
  startAngle: 0,
  endAngle: 2 * Math.PI,
  drawCounterClockwise: false,
};

context.arc(...Object.values(arcConfig));
```